test/unit_tests/autoqasm/test_operators.py

```diff
@@ -932,6 +932,66 @@ def test_list_ops():
     assert test_list_ops.build().to_ir()
 
 
+def test_int_typecasting_on_measure():
+    @aq.main(num_qubits=2)
+    def main():
+        test = int(measure([0, 1]))  # noqa: F841
+
+    expected_ir = """OPENQASM 3.0;
+int[32] test;
+qubit[2] __qubits__;
+bit[2] __bit_0__ = "00";
+__bit_0__[0] = measure __qubits__[0];
+__bit_0__[1] = measure __qubits__[1];
+int[32] __int_1__;
+__int_1__ = __bit_0__;
+test = __int_1__;"""
+    assert main.build().to_ir() == expected_ir
+
+
+def test_int_typecasting_on_string():
+    @aq.main(num_qubits=2)
+    def main():
+        test = int("101", 2)  # noqa: F841
```

Comment:
Should we raise an error, instead?

Reply:
Hello @laurencap, thank you so much for the review! My goal with this test was to have a "normal" call to `int` within a program to check that the built-in `int` function gets called—correctly, with all passed arguments—whenever the first input does not involve a QASM type. Could you explain me how can I change it?

Reply:
I believe this is working as intended, but maybe the confusion could be cleared up by changing the name of the test function to make it more clear what is being tested - e.g. `test_int_typecasting_on_python_string_not_captured`